sdk/include/opentelemetry/sdk/metrics/meter_provider.h

```diff
@@ -0,0 +1,102 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+#ifndef ENABLE_METRICS_PREVIEW
+#  include <memory>
+#  include <mutex>
+#  include <vector>
+#  include "opentelemetry/metrics/meter_provider.h"
+#  include "opentelemetry/nostd/shared_ptr.h"
+#  include "opentelemetry/sdk/metrics/meter.h"
+#  include "opentelemetry/sdk/metrics/meter_context.h"
+#  include "opentelemetry/sdk/resource/resource.h"
+#  include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace metrics
+{
+class MeterProvider final : public opentelemetry::metrics::MeterProvider
+{
+public:
+  /**
+   * Initialize a new meter provider
+   * @param exporters The span exporters for this meter provider
+   * @param readers The readers for this meter provider.
+   * @param views The views for this meter provider
+   * @param resource  The resources for this meter provider.
+   */
+  MeterProvider(std::vector<std::unique_ptr<MeterExporter>> &&exporters,
+                std::vector<std::unique_ptr<MeterReader>> &&readers,
+                std::vector<std::unique_ptr<View>> &&views,
+                sdk::resource::Resource resource = sdk::resource::Resource::Create({})) noexcept;
+
+  /**
+   * Initialize a new meter provider with a specified context
+   * @param context The shared meter configuration/pipeline for this provider.
+   */
+  explicit MeterProvider(std::shared_ptr<sdk::metrics::MeterContext> context) noexcept;
+
+  nostd::shared_ptr<opentelemetry::metrics::Meter> GetMeter(
+      nostd::string_view library_name,
```

Comment:
I suggest that we use `name` instead of `library_name`.

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#get-a-meter